apps/system/lockscreen/js/widgets/clock/lockscreen_clock_widget.js

```diff
@@ -27,13 +27,16 @@
     var f = new navigator.mozL10n.DateTimeFormat();
     var _ = navigator.mozL10n.get;
 
+    /* We use the span tag only as a marker to remove the am/pm component of
+     * the string as we don't show it in the lockscreen. */
     var timeFormat = this._timeFormat.replace('%p', '<span>%p</span>');
     var dateFormat = _('longDateFormat');
 
-    var timeHTML = f.localeFormat(now, timeFormat);
+    var timeText =
+      f.localeFormat(now, timeFormat).replace(/\s<span>.*<\/span>/, '');
```

Comment:
The issue here is that in the case of a 12 hours format we don't want to display the AM/PM text but there's no way to do it using ``mozL10n.DateTimeFormat()``. What we did previously was generate a string that wrapped the AM/PM field in a ``<span>`` and then hid that span via CSS code. This was not only fairly convoluted but forced us to use the unsafe ``innerHTML`` attribute. In my patch I use the ``<span>`` tag only to mark the AM/PM text in the string so I can remove it before displaying it. The UI is unaffected, I've verified it both on my device and in the emulator in 12 & 24 hours formats.